src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -169,14 +171,113 @@ private function createRelationsForItem($item, $formattedRelations)
                 }
             } elseif ($relation instanceof HasOne || $relation instanceof MorphOne) {
                 $modelInstance = $relation->updateOrCreate([], $relationDetails['values']);
+            } elseif ($relation instanceof HasMany || $relation instanceof MorphMany) {
+                $relation_values = $relationDetails['values'][$relationMethod];
+                // if relation values are null we can only attach, also we check if we sent a single dimensional array [1,2,3], or an array of arrays: [[1][2][3]]
+                // if is as single dimensional array we can only attach.
+                if ($relation_values === null || count($relation_values) == count($relation_values, COUNT_RECURSIVE)) {
+                    $this->attachManyRelation($item, $relation, $relationDetails, $relation_values);
+                } else {
+                    $this->createManyEntries($item, $relation, $relationMethod, $relationDetails);
+                }
             }
-
             if (isset($relationDetails['relations'])) {
                 $this->createRelationsForItem($modelInstance, ['relations' => $relationDetails['relations']]);
             }
         }
     }
 
+    /**
+     * When using the HasMany/MorphMany relations as selectable elements we use this function to "mimic-sync" in those relations.
+     * Since HasMany/MorphMany does not have the `sync` method, we manually re-create it.
+     * Here we add the entries that developer added and remove the ones that are not in the list.
+     * This removal process happens with the following rules:
+     * - by default Backpack will behave like a `sync` from M-M relations: it deletes previous entries and add only the current ones.
+     * - `force_delete` is configurable in the field, it's `true` by default. When false, if connecting column is nullable instead of deleting the row we set the column to null.
+     * - `fallback_id` could be provided. In this case instead of deleting we set the connecting key to whatever developer gives us.
+     *
+     * @return void
+     */
```

Comment:
Found a bug: **clearing the select does nothing**. I suspect this is in connection with another bug, which will be fixed separately, but wanted to confirm. @pxpm should we fix this now?

![2021-12-21 13 45 09](https://user-images.githubusercontent.com/1032474/146925024-a3a16479-f2d3-4b71-9366-56bf19833d65.gif)

Reply:
Note that if I only remove one item, it will be saved ok (falls back to `fallback_id` or deletes if `force_delete`).

Reply:
Let's make sure to add a unit test for this too, please 🙏 It's an easy use case to forget when we manually test.

Reply:
Hey @tabacitu !

I've added a test for this. In "core" it works as expected. Problem is with select2. (what gets sent when clearing all entries in the "x") I've reported that to you in other comment, and that's why I previously added the hidden inputs. I will have a look if we can fix this in js select2 events.

Reply:
I have a feeling the JS solution will be 20+ lines long, so way more complicated than the hidden input.

Why is this happening for `HasMany` & `MorphMany`... and not for other relationships? I see `BelongsToMany` and `MorphToMany` are also multiple and are using the `relationship_select` field.

Reply:
@tabacitu exactly about this that I wanted to talk to you. 

So here is the problem: 
- BelongsToMany/MorphMany assumes that when not present in the request they should be deleted. You can't use disabled fields for this relationships otherwise when submitting the form it would clear the relationship. 

- HasMany/MorphMany are created as part of other function where we excude fields that are not present in the form input.
    - So like we know, multiple fields does not post input when empty. 
    - I found a solution that is:
 ```php
.on('select2:unselect', function(e) {
                        if ($multiple && $(this).val().length == 0) {
                            $(this).val(null).trigger('change');
                        }});

// this will send on request:
'entity' => [0 => null]
```
We can check if `entity[0] === null` to know that the field was empty, because when it has value `entity[0]` will never be null unless input is empty.

The reason why it worked with clear and not with the "x" was because "clear" automatically calls the JS value change while we need to manually call when removing in the "x". 

BUT, and I think this is a big BUT: we should not add the empty option `-` on select multiples, and if you remove the empty option from the multiple selects, neither clear or "x" can empty it, because it behaves like a natural select multiple, that will not post the value when empty. 

It should be:
```php
@if ($field['allows_null'] && !$field['multiple'])
            <option value="">-</option>
        @endif
```

To sum up, I think the only reasonable and working solution for this is adding the hidden inputs. that will send a "null" value when the input is cleared, also don't post nothing if input is disabled. 

Let me know if I answered your questions,
Pedro

Reply:
@tabacitu I've created https://github.com/Laravel-Backpack/CRUD/pull/4025

Has this PR merged inside, and the fix (hidden input). Also is pointing to merge here if you agree.

Only did `relationship_select` so you can test it with the current setup in 4.2 demo branch. 

Let me know how I should proceed!

Pedro

Reply:
> BelongsToMany/MorphMany assumes that when not present in the request they should be deleted. You can't use disabled fields for this relationships otherwise when submitting the form it would clear the relationship.

(1) So basically... `BelongsToMany` & `MorphToMany` had a bug about this too? If you set it `disabled` instead of `readonly` it would clear all relations on Save? And the hidden input will solve the problems in `BelongsToMany` & `MorphToMany`, but also in `HasMany` and `MorphMany`? We hit 4 birds with one stone?

> To sum up, I think the only reasonable and working solution for this is adding the hidden inputs. that will send a "null" value when the input is cleared, also don't post nothing if input is disabled.

Yup, I think I agree. Hidden input sounds like the best solution to me too... I'll take a look at the PR right away, but please answer my question above too.

Reply:
@tabacitu we should apply this hidden inputs to ALL selects that are multiple or can be multiple. And it will fix for everything and the behaviour will be the same for everything. 

So technically you kill all birds. I will submit a PR for your review.